.github/workflows/test-e2e.yml

```diff
@@ -51,21 +52,59 @@ jobs:
       - id: prepare-detox-build
         run: RN_SRC_EXT=e2e.ts yarn detox build -c ios.sim.release
         shell: bash
+      - id: upload-detox-build
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.0.3
+        with:
+          name: ItaliaApp.app
+          path: ios/build/Build/Products/Release-iphonesimulator/ItaliaApp.app
+          retention-days: 2
+  run-e2e-tests:    
+    needs: build-detox-app
+    runs-on: macos-latest
+    environment: dev
+    strategy: 
+      fail-fast: false
+      matrix:
+        test: 
+          - "ts/__e2e__/"
+          - "ts/features/bonus/cgn/__e2e__/"
+          - "ts/features/messages/__e2e__/"
+          - "ts/features/euCovidCert/__e2e__/"
+          - "ts/features/wallet/onboarding/__e2e__/"
+    steps:
+      - id: install-applesimutils
+        run: |
+          brew tap wix/brew
+          brew install applesimutils
+        shell: bash
+      - id: checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - id: download-detox-build
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+        with:
+          name: ItaliaApp.app
+          path: ios/build/Build/Products/Release-iphonesimulator/ItaliaApp.app
+      - id: setup
+        uses: ./.github/actions/setup-composite
+        with:
+          use-cache: 'true'
       - id: checkout-dev-server
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
         with:
           repository: pagopa/io-dev-api-server
           path: './_io-dev-api-server_'
       - id: run-e2e-tests
-        run: bash ./.github/scripts/run-e2e-tests.sh
+        run: bash ./.github/scripts/run-e2e-tests.sh ${{ matrix.test }}
       - id: notify-test-failure
         if: failure()
         uses: ./.github/actions/notify-e2e
         env:
+          TEST: ${{ matrix.test }}
           IO_APP_SLACK_HELPER_BOT_TOKEN: ${{ secrets.IO_APP_SLACK_HELPER_BOT_TOKEN }}
       - id: upload-artifacts
-        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v.3.1.2
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v.3.1.2        
         if: always()
         with:
-          name: detox-artifacts
-          path: /tmp/e2e_artifacts/
\ No newline at end of file
+          name: detox-artifacts-${{ matrix.test }}
```

Comment:
If we want a different artifact name for every e2e we need a safer way to handle it's name.

Reply:
We could provide only [`include`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations) values where we set `test` and it's `name` or extract the name from the `test` value (e.g. `detox-artifacts-ts/features/wallet/onboarding/__e2e__/`) by using a run step where we get it with a bash script. Alternatively we can use the [`format`](https://docs.github.com/en/actions/learn-github-actions/expressions) expression to replace `/` with something else. What do you think?